contracts/LendingPoolPeripheral.vy

```diff
@@ -619,140 +758,132 @@ def deprecate():
 
 
 @external
-def deposit(_amount: uint256):
+def depositWeth(_amount: uint256):
+
     """
-    @notice Deposits the given amount in the lending pool
+    @notice Deposits the given amount of WETH in the lending pool
     @dev Logs the `Deposit` event
     @param _amount Value to deposit in wei
     """
 
-    assert not self.isPoolDeprecated, "pool is deprecated, withdraw"
-    assert self.isPoolActive, "pool is not active right now"
-    assert _amount > 0, "_amount has to be higher than 0"
-    assert ILiquidityControls(self.liquidityControlsContract).withinPoolShareLimit(
-        msg.sender,
-        _amount,
-        self,
-        self.lendingPoolCoreContract,
-        self._theoreticalMaxFundsInvestable(_amount)
-    ), "max pool share surpassed"
     assert self._fundsAreAllowed(msg.sender, self.lendingPoolCoreContract, _amount), "not enough funds allowed"
+    self._deposit(_amount, msg.sender)
 
-    if self.whitelistEnabled and not self.whitelistedAddresses[msg.sender]:
-        raise "msg.sender is not whitelisted"
 
-    if not self.isPoolInvesting and self._poolHasFundsToInvestAfterDeposit(_amount):
-        self.isPoolInvesting = True
 
-        log InvestingStatusChanged(
-            self.erc20TokenContract,
-            True,
-            self.erc20TokenContract
-        )
+@external
+@payable
+def depositEth():
 
-    lockPeriodEnd: uint256 = 0
-    lockPeriodAmount: uint256 = 0
-    lockPeriodEnd, lockPeriodAmount = self._computeLockPeriod(msg.sender, _amount)
+    """
+    @notice Deposits the sent amount in the lending pool
+    @dev Logs the `Deposit` event
+    """
 
-    if not ILendingPoolCore(self.lendingPoolCoreContract).deposit(msg.sender, _amount):
-        raise "error creating deposit"
+    _amount: uint256 = msg.value
 
-    ILendingPoolLock(self.lendingPoolLockContract).setInvestorLock(msg.sender, lockPeriodAmount, lockPeriodEnd)
+    log PaymentReceived(msg.sender, msg.sender, _amount)
 
-    log Deposit(msg.sender, msg.sender, _amount, self.erc20TokenContract)
+    self._wrap_and_approve(self.lendingPoolCoreContract, _amount)
+    self._deposit(_amount, self)
```

Comment:
We're storing the value of `msg.value` as a tmp variable, which costs gas. Is there a reason for this instead of just using the `msg.value` where needed?